### PR TITLE
Feat: confirmation prompts

### LINKED ISSUE #686: Confirmation prompts
Some scripts can modify the user's environment without privilege escalation. Adding a confirmation prompt make unrecoverable, unintended changes less likely.

See https://github.com/ChrisTitusTech/linutil/pull/683#issuecomment-2379736118

### PR DESCRIPTION
**Pull request by cartercanedy**
# Add confirmation prompt to prevent accidental command execution

## Type of Change
- [x] New feature
- [x] UI/UX improvement

## Description
Having an accidental keypress result in a significant environment change sucks. Adding a prompt to confirm the execution of a command makes it less likely that a user will have a destructive command accidentally execute and have to decipher the command source to revert the changes (if they even are revertible).

## Testing
locally, tested aborting and confirming command selections

## Impact
should prevent accidental neovim config deletions, at least :)

## Issues / other PRs related
- Resolves #686 

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Comment by lj3954** [comment]
Confirmation prompt doesn't need its own variant, we can use the generic float variant with slight refactoring. https://github.com/lj3954/linutil/commit/0a17f4272397964719f73d1cba2a06eac00c63e0

**Comment by cartercanedy** [comment]
I think the focus specialization is fine there, it prevents us from needing to polute the `FloatContent` interface for other implementors

**Comment by lj3954** [comment]
> I think the focus specialization is fine there, it prevents us from needing to polute the `FloatContent` interface for other implementors

I don't think it's "polluting" the trait. All that needs to be changed is to return an enum rather than a boolean. It's probably more clear that way, since the variants can explicitly state what action is occuring ('None', 'Close', etc) rather than having to look at comments in the trait to explain what each boolean value corresponds to.

**Review by adamperkowski — APPROVED** [review]
Works as expected, no issues.

NIT:
We could add some colors or other "aesthetic" things (like the one i mentioned below for example) to the popup though.

When selecting more than 9 entries, the list looks something like this:
```
1. Alacritty         
2. Android Debloater 
3. Bash Prompt       
4. Bottles           
5. DWM-Titus         
6. Docker            
7. Fastfetch         
8. Grub Theme        
9. Kitty             
10. Linutil Installer
11. Office Suite     
12. ZSH Prompt
```
We could make it check for the number of selected entries and add an extra space after the number for every 9/10 so it'd look like this:
```
9.  Kitty             
10. Linutil Installer
```

**Comment by cartercanedy** [comment]
@lj3954 sorry for jumping to conclusions, I do like what you did with `FloatEvent` and making the `handle_key_event` api more fluent, but I do think that the confirmation float and potentially other potential features could benefit from specialization of `Float` for concrete types instead of being limited by `FloatContent` trait objects

**Comment by cartercanedy** [comment]
how about
```
 1. ...
 ...
10. ...
```

**Comment by adamperkowski** [comment]
> how about
> 
> ```
>  1. ...
>  ...
> 10. ...
> ```

Looks alright.

File: tui/src/confirmation.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        let max_count_str = format!("{}", names.len());
+        let names = names
+            .iter()
+            .zip(1..)
+            .map(|(name, n)| {
+                let count_str = format!("{n}");
+                let space_str = (0..(max_count_str.len() - count_str.len()))
+                    .into_iter()
+                    .map(|_| ' ')
+                    .collect::<String>();
```
Comment (adamperkowski) [inline tui/src/confirmation.rs:35]: Clippy is mad again.

```suggestion
                let space_str = (0..(max_count_str.len() - count_str.len()))
                    .map(|_| ' ')
                    .collect::<String>();
```

Comment (cartercanedy) [reply tui/src/confirmation.rs]: yeah, I should just add `cargo clippy` to a precommit hook at this point

Comment (adamperkowski) [reply tui/src/confirmation.rs]: Add `cargo clippy -- -Dwarnings` so it'll return `0` on warnings.

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]
love it

**Pull request merged**